app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -47,4 +63,187 @@ export default class RnsResolver extends RnsJsDelegate {
         }).catch(error => reject(error));
     });
   }
+
+  /**
+   * Gets all details of a given domain name
+   * @param domainName the domain name to check (without resolver).
+   * @returns {Promise<boolean>} true if it can get all the details correctly, false otherwise.
+   * @param domainName
+   * @returns {Promise<unknown>}
+   */
+  getDomainDetails (domainName) {
+    const domainNameWSuffix = domainName + '.rsk';
+    return new Promise((resolve, reject) => {
+      const getDomainAddress = this.getDomainAddress(domainNameWSuffix);
+      const content = this.getContent(domainNameWSuffix);
+      const expiration = this.getExpirationRemaining(domainNameWSuffix);
+      const getOwner = this.getOwner(domainNameWSuffix);
+      const getResolver = this.getResolver(domainNameWSuffix);
+      Promise.all([getDomainAddress, content, expiration, getOwner, getResolver]).then(values => {
+        let expirationDate = new Date();
+        expirationDate.setDate(expirationDate.getDate() + values[2]);
+        let status = this.getStatus(values[2]);
+        resolve(new DomainDetails(domainNameWSuffix, values[0], values[1], getDateFormatted(expirationDate), false, values[3], status, values[4], false, false));
+      }).catch(error => {
+        reject(error);
+      });
+    });
+  }
+
+  /**
+   * Gets a resolver address if the domain has one
+   * @param domainName with the .rsk extension
+   * @returns {Promise<unknown>}
+   */
+  getResolver(domainName) {
+    return new Promise((resolve, reject) => {
+      this.call(this.rnsContractInstance, 'resolver', [namehash.hash(domainName)]).then(result => {
+        /* TODO: Rodrigo
```

Comment:
What about this TODO? The UI logic includes the case when there is no resolver set?

Reply:
Yep, i forget to delete this!